### ISSUE TITLE
What to do when images need display:block?

### ISSUE THREAD

**Issue by nottrobin** [vanilla-framework/vanilla-framework#1313]
`img`s are `display: inline` by default. This causes problems whenever these images are not directly in the flow of text, because the spaces either side of the `img` element will add to the spacing. This often manifests as an unspecified amount of space above or below the image. To solve this in pure vanilla-framework, you can float the image one way or the other which forces it to `display: block`:

https://github.com/canonical-websites/snapcraft-flask/pull/23/files#r136910513

Could we simply make all image display:block, or otherwise this seems a reasonably case for a utility class.

**Comment by anthonydillon** [comment]
I do not know what unforseen consecenses setting all images too display
block may cause.

A good first start would be to implement the change and see what Percy
picks up. If that is ok, then test on a site or two.

Does anyone know of any reason this might be bad?

**Comment by matthewpaulthomas** [comment]
All the images on Vanilla sites, that I’ve thought might be inline, have turned out not to be. (Though [one is inaccessible as a result](https://github.com/canonical-websites/build.snapcraft.io/issues/957).)

But here are four examples of inline images from current wireframes:

1. A GitHub logo as part of a link to a snap’s GitHub repo.

   ![vanilla-1313-github](https://user-images.githubusercontent.com/19801137/30541797-b96791da-9c74-11e7-8315-1359da374ae9.png)

2. Edit buttons for individual snap details.

   ![vanilla-1313-edit-buttons](https://user-images.githubusercontent.com/19801137/30542420-43781dc0-9c77-11e7-8728-d9e8a270b131.png)

3. Clickable “Open in Snap Directory” buttons.

   ![vanilla-1313-directory-buttons](https://user-images.githubusercontent.com/19801137/30542501-8c410ac6-9c77-11e7-8930-ffd3f1eb592c.png)

4. Color swatches in chart keys.

   ![vanilla-1313-chart-key](https://user-images.githubusercontent.com/19801137/30542614-f55484e8-9c77-11e7-8bb7-b14dc29df45a.png)

**Comment by nottrobin** [comment]
Yeah I think it's safer to make `display-block` only be applied specifically, as it's not the browser default. So I think it should be a utility class.

**Comment by pmahnke** [comment]
reference bug, the work will be picked up
